### ISSUE TITLE
inputs.exec PowerShell Expression = No results

### ISSUE THREAD

**Issue by krushmike** [influxdata/telegraf#3734]
## Bug report
```
[[inputs.exec]]
	commands = [
		"powershell -command \"Get-ClusterResource | Select Name, State | ConvertTo-Json\"",
		"powershell -command \"Get-ClusterResource | Select Name, @{Name='State';Expression={$_.State.ToString()}} | ConvertTo-Json\""
	]	
	timeout = "5s"
	data_format = "json"
	
	tag_keys = [
		"Name"		
	]
```
### System info:

Windows Server 2016 with Failover Cluster and Hyper-V

### Steps to reproduce:

Running Telegraf with the first [[inputs.exec]] command works as expected:

```
"powershell -command \"Get-ClusterResource | Select Name, State | ConvertTo-Json\""
```

> exec,Name=Cluster\ IP\ Address,host=MACHINE1 State=**2** 1517321652000000000
> exec,Name=Cluster\ Name,host=MACHINE1 State=**2** 1517321652000000000
> exec,Name=File\ Share\ Witness,host=MACHINE1 State=**2** 1517321652000000000
> exec,Name=Storage\ Qos\ Resource,host=MACHINE1 State=**2** 1517321652000000000
> exec,Name=Virtual\ Machine\ AEO_POC_00001,host=MACHINE1 State=**2** 1517321652000000000

However, I would like the State to return the actual value, not the enum integer value:

Raw PS command not executed in Telegraf:

```
Get-ClusterResource | Select Name, @{Name='State';Expression={$_.State.ToString()}} | ConvertTo-Json
```
```
[
    {
        "Name":  "Cluster IP Address",
        "State":  "Online"
    },
    {
        "Name":  "Cluster Name",
        "State":  "Online"
    },
    {
        "Name":  "File Share Witness",
        "State":  "Online"
    },
......
]
```
Same command in Telegraf:

```
"powershell -command \"Get-ClusterResource | Select Name, @{Name='State';Expression={$_.State.ToString()}} | ConvertTo-Json\""
```

This returns no output at all when it should be.  Tried lots of different variations of this command without success.

### Expected behavior:

> exec,Name=Cluster\ IP\ Address,host=MACHINE1 State=**Online** 1517321652000000000
> exec,Name=Cluster\ Name,host=MACHINE1 State=**Online**  1517321652000000000
> exec,Name=File\ Share\ Witness,host=MACHINE1 State=**Online**  1517321652000000000
> exec,Name=Storage\ Qos\ Resource,host=MACHINE1 State=**Online**  1517321652000000000
> exec,Name=Virtual\ Machine\ AEO_POC_00001,host=MACHINE1 State=**Online**  1517321652000000000

### Actual behavior:

No results when running telegraph.exe --test with configuration provided above.

**Comment by krushmike** [comment]
It appears Telegraf does not like the .ToString() in the Expression.  If I drop that, I get some output, but obviously State is still an integer.  Tried doing the conversion another way without luck (and this works from the command line):

"powershell -command \"Get-ClusterResource | Select Name, @{Name='State';Expression={[String]$_.State}} | ConvertTo-Json\""

**Comment by danielnelson** [comment]
Probably related to #2230, as a workaround you could avoid passing the args through the command line by using a wrapper script.

**Comment by krushmike** [comment]
Thanks for looking at this @danielnelson.  I think #2230 may definitely be related and I did consider a separate script file.  Considering how simple this script is, basically a one liner, I did not expect issues with Telegraph digesting it.  Obviously, it is easier to provide a one liner to Telegraph than manage separate script files, but using separate is a suitable workaround.

It is interesting not to see any parsing errors in the Telegraf log file as well.  There is simply no output...

In any case, hopefully there will be a fix for this in the near future.
